app/frontend/wallet/wallet.ts

```diff
@@ -0,0 +1,118 @@
+import {Account} from './account'
+import BlockchainExplorer from './blockchain-explorer'
+
+const Wallet = ({config, cryptoProvider}) => {
+  const blockchainExplorer = BlockchainExplorer(config)
+
+  const accounts: Array<ReturnType<typeof Account>> = []
+
+  function loadAccount(accountIndex: number) {
+    const newAccount = Account({
+      config,
+      cryptoProvider,
+      blockchainExplorer,
+      accountIndex,
+    })
+    accounts.push(newAccount)
+  }
+
+  async function discoverAccounts() {
+    let shouldExplore = true
+    let accountIndex = 0
+    while (shouldExplore && !accounts[accountIndex]) {
+      const newAccount = Account({
+        config,
+        cryptoProvider,
+        blockchainExplorer,
+        accountIndex,
+      })
+      const isAccountUsed = await newAccount.isAccountUsed()
+      if (isAccountUsed || accountIndex === 0) accounts.push(newAccount)
```

Comment:
why not just call loadAccount(index) repeatedly until the last account in the accounts list is unused/index is equal to 100? Wouldn't that yield the same result, or am I missing some nuance here?